File: libs/document-processor-next/src/registry/contract-processor-registry-builder.ts
import { ContractProcessorRegistry } from './contract-processor-registry.js';
import { MyOSTimelineChannelProcessor } from './processors/myos-timeline-channel-processor.js';
import { TimelineChannelProcessor } from './processors/timeline-channel-processor.js';
import { SequentialWorkflowHandlerProcessor } from './processors/sequential-workflow-processor.js';
import { OperationMarkerProcessor } from './processors/operation-marker-processor.js';
import { SequentialWorkflowOperationProcessor } from './processors/sequential-workflow-operation-processor.js';
import { AnyContractProcessor } from './types.js';

export class ContractProcessorRegistryBuilder {
  private constructor(private readonly registry: ContractProcessorRegistry) {}

  static create(): ContractProcessorRegistryBuilder {
    return new ContractProcessorRegistryBuilder(
      new ContractProcessorRegistry(),
    );
  }

  registerDefaults(): ContractProcessorRegistryBuilder {
    this.registry.register(new MyOSTimelineChannelProcessor());
    this.registry.register(new TimelineChannelProcessor());
    this.registry.register(new SequentialWorkflowHandlerProcessor());
    this.registry.register(new OperationMarkerProcessor());
    this.registry.register(new SequentialWorkflowOperationProcessor());
    return this;
  }

  register(processor: AnyContractProcessor): ContractProcessorRegistryBuilder {
    this.registry.register(processor);
    return this;
  }

  build(): ContractProcessorRegistry {
    return this.registry;
  }
}

File: references/java-sdk/src/main/java/blue/language/processor/ContractProcessorRegistryBuilder.java
package blue.language.processor;

import blue.language.processor.model.Contract;
import blue.language.processor.registry.processors.CompositeTimelineChannelProcessor;
import blue.language.processor.registry.processors.MyOSTimelineChannelProcessor;
import blue.language.processor.registry.processors.OperationMarkerProcessor;
import blue.language.processor.registry.processors.SequentialWorkflowHandlerProcessor;
import blue.language.processor.registry.processors.SequentialWorkflowOperationProcessor;
import blue.language.processor.registry.processors.TimelineChannelProcessor;

import java.util.Objects;

/**
 * Builder utility concentrated around contract processor registration.
 */
public final class ContractProcessorRegistryBuilder {

    private final ContractProcessorRegistry registry;

    private ContractProcessorRegistryBuilder(ContractProcessorRegistry registry) {
        this.registry = registry;
    }

    public static ContractProcessorRegistryBuilder create() {
        return new ContractProcessorRegistryBuilder(new ContractProcessorRegistry());
    }

    public ContractProcessorRegistryBuilder registerDefaults() {
        registry.registerChannel(new TimelineChannelProcessor());
        registry.registerChannel(new CompositeTimelineChannelProcessor());
        registry.registerChannel(new MyOSTimelineChannelProcessor());
        registry.registerHandler(new SequentialWorkflowHandlerProcessor());
        registry.registerHandler(new SequentialWorkflowOperationProcessor());
        registry.registerMarker(new OperationMarkerProcessor());
        return this;
    }

    public ContractProcessorRegistryBuilder register(ContractProcessor<? extends Contract> processor) {
        Objects.requireNonNull(processor, "processor");
        registry.register(processor);
        return this;
    }

    public ContractProcessorRegistry build() {
        return registry;
    }
}
